src/testdir/test_startup.vim

```diff
@@ -508,38 +508,6 @@ func Test_font()
   call delete('Xtest_font')
 endfunc
 
-" Test the -geometry argument (for GUI only).
-func Test_geometry()
-  CheckCanRunGui
-  CheckAnyOf Feature:gui_gtk Feature:gui_motif Feature:gui_athena
-
-  if has('gui_motif') || has('gui_athena')
-    " FIXME: With GUI Athena or Motif, the value of getwinposx(),
-    "        getwinposy() and getwinpos() do not match exactly the
-    "        value given in -geometry. Why?
-    "        So only check &columns and &lines for those GUIs.
-    let after =<< trim [CODE]
-      call writefile([&columns, &lines], "Xtest_geometry")
-      qall
-    [CODE]
-    if RunVim([], after, '-f -g -geometry 31x13+41+43')
-      let lines = readfile('Xtest_geometry')
-      call assert_equal(['31', '13'], lines)
-    endif
-  else
-    let after =<< trim [CODE]
-      call writefile([&columns, &lines, getwinposx(), getwinposy(), string(getwinpos())], "Xtest_geometry")
-      qall
-    [CODE]
-    if RunVim([], after, '-f -g -geometry 31x13+41+43')
-      let lines = readfile('Xtest_geometry')
-      call assert_equal(['31', '13', '41', '43', '[41, 43]'], lines)
-    endif
-  endif
-
-  call delete('Xtest_geometry')
-endfunc
```

Comment:
If the test expectations are not reliable, how about we instead relax the expectation (maybe part of it is correct) or maybe even remove the expectation (i.e. remove the `assert_equal(...)`) but we still run the test, so that:
* code is covered with tests
* and we merely check that it does not crash

Reply:
Perhaps we should instead test, that the lines and columns match?

Reply:
> If the test expectations are not reliable, how about we instead relax the expectation ...

I'd be fine with that. I looked around the test suite for other tests that were meant to exercise code without enforcing any kind of post condition and didn't see any so I suspected it might be unwelcome. 


> Perhaps we should instead test, that the lines and columns match?

What frustrates this most is that the initial window size is properly based on the columns and lines given in the `-geometry` value. Later (but still before we can test) other window elements are realized (e.g. menu bar, toolbar, titlebar) causing the application to be notified of the updated dimensions. The `columns` and `lines` values are updated to reflect the new dimensions as if the user resized the window. I did some investigation to determine whether the manual user resize events could be differentiated from the resize events from the other window elements squeezing the area available for the text. I believe the proper way (from the GTK team's point of view) is to use the form window (that displays the edited text) as GTK's "geometry" widget. However, [to quote from the current code](https://github.com/vim/vim/blob/master/src/gui_gtk_x11.c#L3047-L3049):

```
	// Using gui.formwin as geometry widget doesn't work as expected
	// with GTK+ 2 -- dunno why.  Presumably all the resizing hacks
	// in Vim confuse GTK+.
```

The best alternative test I could come up with is running vim twice with significantly different geometries and testing that they differ in some ways. For example, I can't identify a situation in which two differently sized window allocations are requested and the larger request receives and area smaller than the smaller request.

Reply:
> >    If the test expectations are not reliable, how about we instead relax the expectation ...

> I'd be fine with that. I looked around the test suite for other tests that were meant to exercise code without enforcing any kind of post condition and didn't see any so I suspected it might be unwelcome.

At least in `test_hardcopy.vim`, I see such comment `" Only test that this doesn't throw an error.` with a test that does not have any `assert_*()`.  I find it useful to exercise all the code if possible, as sometimes running with valgrind or asan we can find leaks or memory errors. Of course it's not ideal to lack assertions, but it's still better than not running the code at all.

Reply:
If the problem is with menu and toolbar, we can set guioptions to exclude them.
Removing the whole test is not a good idea.

Reply:
@brammool the toolbar and menu widgets are involved in the current expression of this issue but setting `guioptions` to exclude them doesn't overcome the issue totally and does not address the root issue. In GTK3, because we use the main window as the geometry window, we're essentially offering part of the `lines` geometry to the titlebar. That's not the case in GTK2 due to the difference in client-side decorations.

The root issue is that the `-geometry` flag can request a specific size and the current implementation passes that request through to GTK and X but the current code is pretty explicit about accepting a different size. That is to say that the current test is very specifically tuned to the Xvfb experience and assumes that behavior that is completely valid in the wild will not happen. For example, this test could also fail if we started a window manager within the test suite even if we did not change the window sizing code.

I'll update the PR later today with a change to induce coverage of the initial geometry code while loosening the passing criteria.